pkg/registryclient/client.go

```diff
@@ -94,6 +98,9 @@ func New(options ...Option) (Client, error) {
 	if cfg.tracing {
 		c.transport = tracing.Transport(cfg.transport, otelhttp.WithFilter(tracing.RequestFilterIsInSpan))
 	}
+	if cfg.transport.TLSClientConfig != nil && cfg.transport.TLSClientConfig.InsecureSkipVerify {
+		c.allowInsecureRegistry = true
+	}
```

Comment:
instead of extracting it from transport, you can set a field in the config `allowInsecureRegistry` which will then be set in the client and used by `NameOptions()`

Reply:
Fixed